pkg/cmd/server/kubernetes/master_config.go

```diff
@@ -157,6 +157,33 @@ func BuildKubernetesMasterConfig(options configapi.MasterConfig, requestContextM
 
 		EnableV1Beta3: configapi.HasKubernetesAPILevel(*options.KubernetesMasterConfig, "v1beta3"),
 		DisableV1:     !configapi.HasKubernetesAPILevel(*options.KubernetesMasterConfig, "v1"),
+
+		ExtraServicePorts: []kapi.ServicePort{
```

Comment:
only do this if the master is configured to be a DNS server (`options.DNSConfig != nil`, I think)